libvast/src/segment.cpp

```diff
@@ -11,24 +11,38 @@
  * contained in the LICENSE file.                                             *
  ******************************************************************************/
 
+#include "vast/segment.hpp"
+
 #include <caf/binary_deserializer.hpp>
 
 #include "vast/bitmap.hpp"
 #include "vast/bitmap_algorithms.hpp"
+#include "vast/detail/assert.hpp"
+#include "vast/detail/byte_swap.hpp"
+#include "vast/detail/narrow.hpp"
 #include "vast/ids.hpp"
 #include "vast/logger.hpp"
-#include "vast/segment.hpp"
 #include "vast/si_literals.hpp"
 #include "vast/table_slice.hpp"
 
-#include "vast/detail/assert.hpp"
-#include "vast/detail/byte_swap.hpp"
-#include "vast/detail/narrow.hpp"
-
 namespace vast {
 
 using namespace binary_byte_literals;
 
+ids segment::meta_data::flat_slice_ids() const {
+  ids result;
+  auto concat = [&](const ids& x) { result |= x; };
```

Comment:
This seems rather heavy-weight. Why not just append bits to `result`?

Reply:
Can you explain what's heavyweight about this and how I would achieve semantically identical but more efficient code using append member functions?

Reply:
Nevermind, `visit_ids` gets the axe anyways.